### PR TITLE
Inference tests

### PR DESCRIPTION
**Pull request by jennyfolkesson**
Sorry for the massiveness of this PR, I had to rewrite the inference and its dataset class to understand what was happening. Now it works for the microglial data at least.
I'll create a separate PR for testing 2.5 and 3D, this PR got a bit out of hand as it is.

### CODE REVIEW

File: micro_dl/inference/image_inference.py [hunk 1]
```diff
@@ -1,230 +1,240 @@
...
-        if image_param_dict['image_format'] == 'zyx':
-            z_dim = 2 if self.data_format == 'channels_first' else 1
-        elif image_param_dict['image_format'] == 'xyz':
-            z_dim = 4 if self.data_format == 'channels_first' else 3
-
-        tile_option = None
-        if 'num_slices' in vol_inf_dict and vol_inf_dict['num_slices'] > 1:
-            tile_option = 'tile_z'
+        if self.image_format == 'zyx':
+            self.z_dim = 2 if self.data_format == 'channels_first' else 1
```
Comment (pranathivemuri) [inline micro_dl/inference/image_inference.py]: Is there the other option channels_last?

File: micro_dl/config_inference.yml [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+images:
     image_format: 'zyx'
     im_ext: '.npy'
     crop_shape: [96, 2048, 2048]
-metrics: ['r2', 'corr', 'ssim']
-metrics_orientations: ['xy', 'xyz', 'xz', 'yz']
-mask_params_dict:
+metrics:
+    metrics: ['r2', 'corr', 'ssim']
+    metrics_orientations: ['xy', 'xyz', 'xz', 'yz']
```
[the remark this answers is not in the document]
Comment (jennyfolkesson) [reply micro_dl/config_inference.yml]: Good point - I'll add a description in inference/evaluation_metrics.py and refer to that in image_inference.py

File: micro_dl/inference/image_inference.py [hunk 3]
```diff
@@ -1,230 +1,240 @@
...
+              inference on a center sub volume.
+             list tile_shape: shape of tile for tiling along xyz.
+             int/list num_overlap: int for tile_z, list for tile_xyz
+             str overlap_operation: e.g. 'mean'
+        :param int gpu_id: GPU number to use. -1 for debugging (no GPU)
+        :param float/None gpu_mem_frac: Memory fractions to use corresponding
          to gpu_ids
-        :param list metrics_list: list of metrics to estimate. available
-         metrics: [ssim, corr, r2, mse, mae}]
          TODO: add accuracy and dice coeff to metrics list
```
[the remark this answers is not in the document]
Comment (jennyfolkesson) [reply micro_dl/inference/image_inference.py]: They need to be added to the class in inference/evaluation_metrics.py...

File: micro_dl/utils/preprocess_utils.py [hunk 4]
```diff
@@ -6,23 +6,26 @@
...
 
-    :param dict pp_config: Preprocessing config
+    :param str mask_dir: Mask directory
+    :param str input_dir: Input image directory, to match masks with images
+    :param int/None mask_channel: Channel idx assigned to masks
     :return int mask_channel: New channel index for masks for writing tiles
-    :raises AssertionError: If 'masks' in pp_config contains both channels
-        and mask_dir (the former is for generating masks from a channel)
-    :raises IOError: If no csv file is present in mask_dir
+    :raises IOError: If not just one csv file is present in mask_dir
```
[the remark this answers is not in the document]
Comment (jennyfolkesson) [reply micro_dl/utils/preprocess_utils.py]: Good catch!

**Comment by pranathivemuri** [comment]
@jennyfolkesson thanks for addressing my comments. please feel free to merge if you think it's done

**Pull request merged**